packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -1,193 +1,63 @@
- * thanks to the 'optionText' and 'optionValue' attributes.
- * @example
- * const choices = [
- *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
- *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
- * ];
- * <AutocompleteInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
- *
- * `optionText` also accepts a function, so you can shape the option text at will:
- * @example
- * const choices = [
- *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
- *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
- * ];
- * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
- * <AutocompleteInput source="author_id" choices={choices} optionText={optionRenderer} />
- *
- * `optionText` also accepts a React Element, that will be cloned and receive
- * the related choice as the `record` prop. You can use Field components there.
- * Note that you must also specify the `matchSuggestion` prop
- * @example
- * const choices = [
- *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
- *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
- * ];
- * const matchSuggestion = (filterValue, choice) => choice.first_name.match(filterValue) || choice.last_name.match(filterValue);
- * const FullNameField = ({ record }) => <span>{record.first_name} {record.last_name}</span>;
- * <SelectInput source="gender" choices={choices} optionText={<FullNameField />} matchSuggestion={matchSuggestion} />
- *
- * The choices are translated by default, so you can use translation identifiers as choices:
- * @example
- * const choices = [
- *    { id: 'M', name: 'myroot.gender.male' },
- *    { id: 'F', name: 'myroot.gender.female' },
- * ];
- *
- * However, in some cases (e.g. inside a `<ReferenceInput>`), you may not want
- * the choice to be translated. In that case, set the `translateChoice` prop to false.
- * @example
- * <AutocompleteInput source="gender" choices={choices} translateChoice={false}/>
- *
- * The object passed as `options` props is passed to the material-ui <TextField> component
- *
- * @example
- * <AutocompleteInput source="author_id" options={{ color: 'secondary', InputLabelProps: { shrink: true } }} />
- */
 export const AutocompleteInput = (props: AutocompleteInputProps) => {
     const {
         allowEmpty,
-        className,
-        classes: classesOverride,
-        clearAlwaysVisible,
-        choices = [],
+        choices,
+        create,
         createLabel,
         createItemLabel,
         createValue,
-        create,
-        disabled,
         emptyText,
         emptyValue,
         format,
-        fullWidth,
         helperText,
         id: idOverride,
         input: inputOverride,
-        inputText,
         isRequired: isRequiredOverride,
         label,
-        limitChoicesToValue,
         loaded,
         loading,
-        margin = 'dense',
+        limitChoicesToValue,
```

Comment:
This one needs documentation

Reply:
I actually migrated it without thinking but I fail to see its usefulness

Reply:
I'll revisit once the AutocompleteArrayInput is migrated to use Autocomplete